core/src/main/java/cz/o2/proxima/scheme/SchemaDescriptors.java

```diff
@@ -0,0 +1,885 @@
+   */
+  public static ArrayTypeDescriptor<byte[]> bytes() {
+    return bytes(null);
+  }
+
+  /**
+   * Create {@link ArrayTypeDescriptor} for byte array with default value.
+   *
+   * @param defaultValueFactory default value factory
+   * @return Array type descriptor
+   */
+  public static ArrayTypeDescriptor<byte[]> bytes(Factory<byte[]> defaultValueFactory) {
+    return arrays(primitives(AttributeValueType.BYTE), defaultValueFactory);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Integer}.
+   *
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Integer> integers() {
+    return integers(null);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Integer} with default value.
+   *
+   * @param defaultValueFactory default value factory
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Integer> integers(Factory<Integer> defaultValueFactory) {
+    return primitives(AttributeValueType.INT, defaultValueFactory);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Long}.
+   *
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Long> longs() {
+    return longs(null);
+  }
+
+  /**
+   * Create {@link ArrayTypeDescriptor} for {@code Long} with default value.
+   *
+   * @param defaultValueFactory default value factory
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Long> longs(Factory<Long> defaultValueFactory) {
+    return primitives(AttributeValueType.LONG, defaultValueFactory);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Double}.
+   *
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Double> doubles() {
+    return doubles(null);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Double} with default value.
+   *
+   * @param defaultValueFactory default value factory
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Double> doubles(Factory<Double> defaultValueFactory) {
+    return primitives(AttributeValueType.DOUBLE, defaultValueFactory);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Float}.
+   *
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Float> floats() {
+    return floats(null);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Float} with default value.
+   *
+   * @param defaultValueFactory default value factory
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Float> floats(Factory<Float> defaultValueFactory) {
+    return primitives(AttributeValueType.FLOAT, defaultValueFactory);
+  }
+
+  /**
+   * Create {@link PrimitiveTypeDescriptor} for {@code Boolean}.
+   *
+   * @return Primitive type descriptor
+   */
+  public static PrimitiveTypeDescriptor<Boolean> booleans() {
```

Comment:
I don't quite understand the plural here. (as in Beam's TypeDescriptors.longs()). :)
The place where it is the most confusing is when creating field descriptor, because it takes a single parameter `name`, so `SchemaDescriptors.fields(name, ...)` makes no sense to me, because due to the name parameter it cannot create "two" fields, but just one specific field.

Reply:
I was also thinking a lot about using plural or not. On the end the reason is simple - you can't use for example `boolean`, `integer` etc because it's reserved word in java. Feel free to suggest something better, I spend some time on this and can't find something better.

You are right with `fields` => `field`, I'll probably rename it.

Reply:
I'll prefer removing fields() completely (see later), so no problem. Makes sense for the rest.